src/GitVersionCore/Configuration/ConfigurationUtils.cs

```diff
@@ -0,0 +1,181 @@
+                if (branchConfig.Value.IsSourceBranchFor == null) continue;
+                foreach (var isSourceBranch in branchConfig.Value.IsSourceBranchFor)
+                {
+                    config.Branches[isSourceBranch].SourceBranches.Add(branchConfig.Key);
+                }
+            }
+        }
+
+        private static readonly Dictionary<string, int> DefaultPreReleaseWeight =
+            new Dictionary<string, int>
+            {
+                { ConfigurationConstants.DevelopBranchRegex, 0 },
+                { ConfigurationConstants.HotfixBranchRegex, 30000 },
+                { ConfigurationConstants.ReleaseBranchRegex, 30000 },
+                { ConfigurationConstants.FeatureBranchRegex, 30000 },
+                { ConfigurationConstants.PullRequestRegex, 30000 },
+                { ConfigurationConstants.SupportBranchRegex, 55000 },
+                { ConfigurationConstants.MasterBranchRegex, 55000 }
+            };
+        private const IncrementStrategy DefaultIncrementStrategy = IncrementStrategy.Inherit;
+
+        public static void ApplyBranchDefaults(Config config,
+            BranchConfig branchConfig,
+            string branchRegex,
+            List<string> sourceBranches,
+            string defaultTag = "useBranchName",
+            IncrementStrategy? defaultIncrementStrategy = null, // Looked up from main config
+            bool defaultPreventIncrement = false,
+            VersioningMode? defaultVersioningMode = null, // Looked up from main config
+            bool defaultTrackMergeTarget = false,
+            string defaultTagNumberPattern = null,
+            bool tracksReleaseBranches = false,
+            bool isReleaseBranch = false,
+            bool isMainline = false)
+        {
+            branchConfig.Regex = string.IsNullOrEmpty(branchConfig.Regex) ? branchRegex : branchConfig.Regex;
+            branchConfig.SourceBranches = branchConfig.SourceBranches == null || !branchConfig.SourceBranches.Any()
+                ? sourceBranches : branchConfig.SourceBranches;
+            branchConfig.Tag ??= defaultTag;
+            branchConfig.TagNumberPattern ??= defaultTagNumberPattern;
+            branchConfig.Increment ??= defaultIncrementStrategy ?? config.Increment ?? DefaultIncrementStrategy;
+            branchConfig.PreventIncrementOfMergedBranchVersion ??= defaultPreventIncrement;
+            branchConfig.TrackMergeTarget ??= defaultTrackMergeTarget;
+            branchConfig.VersioningMode ??= defaultVersioningMode ?? config.VersioningMode;
+            branchConfig.TracksReleaseBranches ??= tracksReleaseBranches;
+            branchConfig.IsReleaseBranch ??= isReleaseBranch;
+            branchConfig.IsMainline ??= isMainline;
+            DefaultPreReleaseWeight.TryGetValue(branchRegex, out var defaultPreReleaseNumber);
+            branchConfig.PreReleaseWeight ??= defaultPreReleaseNumber;
+        }
+
+        public static void VerifyConfiguration(Config readConfig)
```

Comment:
Can't this be exposed as a `Verify()` method on the `Config` class?

Reply:
Made it as extension to Config, in the ConfigExtensions, renamed to `Verify`